core/src/main/java/org/wso2/testgrid/core/configchangeset/ConfigChangeSetExecutorUnix.java

```diff
@@ -0,0 +1,242 @@
+        String tinkererHost = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_REST_BASE_PATH);
+        // Get authentication detail from config.properties and encode with BASE64
+        String authenticationString = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_USERNAME) + ":" +
+                ConfigurationContext.getProperty(
+                        ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_PASSWORD);
+        String authenticationToken = "Basic " + Base64.getEncoder().encodeToString(
+                authenticationString.getBytes(StandardCharsets.UTF_8));
+        boolean executionStatus = true;
+        String agentLink = tinkererHost + "test-plan" + TestGridConstants.FILE_SEPARATOR + testPlan.getId() +
+                TestGridConstants.FILE_SEPARATOR + "agents";
+        try {
+            // Get list of agent for given test plan id
+            Content agentResponse = Request.Get(agentLink)
+                    .addHeader(HttpHeaders.AUTHORIZATION, authenticationToken).
+                            execute().returnContent();
+
+            Agent[] agents = new Gson().fromJson(agentResponse.asString(), Agent[].class);
+            // Execute command on found agents
+            for (Agent agent : Arrays.asList(agents)) {
+                logger.info("Start sending commands to agent " + agent.getAgentId());
+                for (String shellCommand : initShellCommand) {
+                    logger.info("Execute: " + shellCommand);
+                    Content shellCommandResponse = sendShellCommand(tinkererHost, authenticationToken
+                            , agent, shellCommand);
+                    JsonParser jsonParser = new JsonParser();
+                    JsonObject result = (JsonObject) jsonParser.parse(shellCommandResponse.asString());
+                    // Set default  value as empty
+                    int exitValue = 0;
+                    String code = "";
+                    String response = "";
+                    if (result.get("exitValue") != null) {
+                        exitValue = result.get("exitValue").getAsInt();
+                    }
+                    if (result.get("code") != null) {
+                        code = result.get("code").getAsString();
+                    }
+                    if (result.get("response") != null) {
+                        response = result.get("response").getAsString();
+                    }
+                    logger.info("Agent exit value: " + exitValue);
+                    logger.info("Agent code: " + code);
+                    logger.info("Agent response: \n" + response);
+                    // Agent code default is SHELL, if time out then, it return code as 408
+                    if (code.equals(String.valueOf(HttpStatus.SC_REQUEST_TIMEOUT)) || exitValue != 0) {
```

Comment:
What if the code is HTTP 500? Is it enough to only check for request timeout? 

As a general rule in HTTP status codes, 2xx codes are considered successful, 4xx are client-side errors, and 5xx are server-side errors. In this case, any code other than 2xx code range is an error scenario.

Reply:
This is the timeout response from the agent. Current implementation it only set back timeout status as code back to the test executor. Other responses are not handle. This will be fixed with next feature development.

Reply:
Well, timeout status code may be the only thing hard-coded. But, other error codes can also be present for numerous reasons. For example, if an exception get thrown by the agent rest service, then the tinkerer will just receive a 500 status code. Another example, say the API contract of the agent rest service changed. Then, your HTTP requests may fail with a 404 not found. 
Therefore, it's important to handle other status codes.
We can of course do this in a separate PR, but please open a github ticket with highest priority. :)